x-pack/test/functional/apps/endpoint/host_list.ts

```diff
@@ -71,29 +74,66 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       await testSubjects.existOrFail('hostDetailsUpperList');
       await testSubjects.existOrFail('hostDetailsLowerList');
     });
 
-    it('displays no items found when empty', async () => {
-      // clear out the data and reload the page
-      await esArchiver.unload('endpoint/metadata/api_feature');
-      await pageObjects.common.navigateToUrlWithBrowserHistory('endpoint', '/hosts');
-      // get the table data and verify no entries appear
-      const tableData = await pageObjects.endpoint.getEndpointAppTableData('hostListTable');
-      expect(tableData[1][0]).to.equal('No items found');
-      // reload the data so the other tests continue to pass
-      await esArchiver.load('endpoint/metadata/api_feature');
+    it('update details flyout when new hostname is clicked on', async () => {
+      // display flyout for the first host in the list
+      await (await testSubjects.findAll('hostnameCellLink'))[0].click();
+      await testSubjects.existOrFail('hostDetailsFlyoutTitle');
+      const hostDetailTitle0 = await testSubjects.getVisibleText('hostDetailsFlyoutTitle');
+      // select the 2nd host in the host list
+      await (await testSubjects.findAll('hostnameCellLink'))[1].click();
+      await pageObjects.endpoint.waitForVisibleTextToChange(
+        'hostDetailsFlyoutTitle',
+        hostDetailTitle0
+      );
+      const hostDetailTitle1 = await testSubjects.getVisibleText('hostDetailsFlyoutTitle');
+      expect(hostDetailTitle1).to.not.eql(hostDetailTitle0);
+    });
+
+    it('details flyout remains the same when current hostname is clicked on', async () => {
+      // display flyout for the first host in the list
+      await (await testSubjects.findAll('hostnameCellLink'))[1].click();
+      await testSubjects.existOrFail('hostDetailsFlyoutTitle');
+      const hostDetailTitleInitial = await testSubjects.getVisibleText('hostDetailsFlyoutTitle');
+      // select the same host in the host list
+      await (await testSubjects.findAll('hostnameCellLink'))[1].click();
+      await sleep(1000); // give page time to refresh
+      const hostDetailTitleNew = await testSubjects.getVisibleText('hostDetailsFlyoutTitle');
+      expect(hostDetailTitleNew).to.eql(hostDetailTitleInitial);
+    });
+
+    describe('no data', () => {
+      before(async () => {
+        // clear out the data and reload the page
+        await esArchiver.unload('endpoint/metadata/api_feature');
+        await pageObjects.common.navigateToUrlWithBrowserHistory('endpoint', '/hosts');
+      });
+      after(async () => {
+        // reload the data so the other tests continue to pass
+        await esArchiver.load('endpoint/metadata/api_feature');
+      });
+      it('displays no items found when empty', async () => {
+        // get the table data and verify no entries appear
+        const tableData = await pageObjects.endpoint.getEndpointAppTableData('hostListTable');
+        expect(tableData[1][0]).to.equal('No items found');
```

Comment:
could you maybe use more descriptive variable names over just the indexes? https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#use-array-destructuring

Reply:
Implemented